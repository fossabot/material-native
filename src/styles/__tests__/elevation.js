'use strict';
import elevation from '../elevation';
import {Platform} from 'react-native';

const PlatformOS = Platform.OS;
beforeEach(() => {
	Platform.OS = PlatformOS;
});
afterEach(() => {
	Platform.OS = PlatformOS;
});

describe('Android', () => {
	for ( const n of [0, 1, 2, 3, 4, 6, 8, 9, 12, 16, 24] ) {
		test(`elevation(${n}) matches snapshot`, () => {
			Platform.OS = 'android';
			expect(elevation(n)).toMatchSnapshot();
		});
	}
});

describe('iOS', () => {
	for ( const n of [0, 1, 2, 3, 4, 6, 8, 9, 12, 16, 24] ) {
		test(`elevation(${n}) matches snapshot`, () => {
			Platform.OS = 'ios';
			expect(elevation(n)).toMatchSnapshot();
		});
	}
});
