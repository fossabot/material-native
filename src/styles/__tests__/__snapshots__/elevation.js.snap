// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android elevation(0) matches snapshot 1`] = `
Object {
  "elevation": 0,
}
`;

exports[`Android elevation(1) matches snapshot 1`] = `
Object {
  "elevation": 1,
}
`;

exports[`Android elevation(2) matches snapshot 1`] = `
Object {
  "elevation": 2,
}
`;

exports[`Android elevation(3) matches snapshot 1`] = `
Object {
  "elevation": 3,
}
`;

exports[`Android elevation(4) matches snapshot 1`] = `
Object {
  "elevation": 4,
}
`;

exports[`Android elevation(6) matches snapshot 1`] = `
Object {
  "elevation": 6,
}
`;

exports[`Android elevation(8) matches snapshot 1`] = `
Object {
  "elevation": 8,
}
`;

exports[`Android elevation(9) matches snapshot 1`] = `
Object {
  "elevation": 9,
}
`;

exports[`Android elevation(12) matches snapshot 1`] = `
Object {
  "elevation": 12,
}
`;

exports[`Android elevation(16) matches snapshot 1`] = `
Object {
  "elevation": 16,
}
`;

exports[`Android elevation(24) matches snapshot 1`] = `
Object {
  "elevation": 24,
}
`;

exports[`iOS elevation(0) matches snapshot 1`] = `Object {}`;

exports[`iOS elevation(1) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 1,
}
`;

exports[`iOS elevation(2) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 2,
}
`;

exports[`iOS elevation(3) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 3,
}
`;

exports[`iOS elevation(4) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 4,
}
`;

exports[`iOS elevation(6) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 6,
}
`;

exports[`iOS elevation(8) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 8,
}
`;

exports[`iOS elevation(9) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 9,
}
`;

exports[`iOS elevation(12) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 12,
}
`;

exports[`iOS elevation(16) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 16,
}
`;

exports[`iOS elevation(24) matches snapshot 1`] = `
Object {
  "shadowColor": "#000",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.3,
  "shadowRadius": 24,
}
`;
